src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -0,0 +1,146 @@
+import { RemoteGenerated } from "../../../crd";
+import { anywhere } from "./anywhere";
+import { UDSConfig } from "../../../../config";
+
+const log = setupLogger(Component.OPERATOR_GENERATORS);
+
+// Maintain a set of all node internal IPs
+const nodeSet = new Set<string>();
+
+/**
+ * Initialize the node targets by fetching the current nodes in the cluster
+ * and populating the nodeSet with their Internal IPs.
+ */
+export async function initAllNodesTarget() {
+  try {
+    const nodes = await fetchKubernetesNodes();
+    nodeSet.clear();
+
+    for (const node of nodes.items) {
+      const ip = getNodeInternalIP(node);
+      if (ip) nodeSet.add(ip);
+    }
+  } catch (err) {
+    log.error("error fetching node IPs:", err);
+  }
+}
+
+/**
+ * Returns the egress CIDRs of all known nodes as network policy peers.
+ * If none are known, defaults to 0.0.0.0/0 and logs a warning.
+ */
+export function nodeCIDRs(): V1NetworkPolicyPeer[] {
+  const policies = buildNodePolicies([...nodeSet]);
+  if (policies.length > 0) return policies;
+
+  log.warn("Unable to get API server CIDR, defaulting to 0.0.0.0/0");
+  return [anywhere];
+}
+
+/**
+ * When a node is created or updated, if it's Ready, add its IP to the set,
+ * rebuild the policies, and update the NetworkPolicies.
+ */
+export async function updateKubeNodesFromCreateUpdate(node: kind.Node) {
+  const isReady = node.status?.conditions?.some(
+    (condition: V1NodeCondition) => condition.type === "Ready" && condition.status === "True",
+  );
+
+  if (isReady) {
+    const ip = getNodeInternalIP(node);
+    if (ip) nodeSet.add(ip);
+  }
+
+  const policies = buildNodePolicies([...nodeSet]);
+  await updateKubeNodesNetworkPolicies(policies);
+}
+
+/**
+ * When a node is deleted, remove its IP from the set, rebuild the policies,
+ * and update the NetworkPolicies.
+ */
+export async function updateKubeNodesFromDelete(node: kind.Node) {
+  const ip = getNodeInternalIP(node);
+  if (ip) nodeSet.delete(ip);
+
+  const policies = buildNodePolicies([...nodeSet]);
+  await updateKubeNodesNetworkPolicies(policies);
+}
+
+/**
+ * Fetch all Kubernetes nodes.
+ */
+async function fetchKubernetesNodes(): Promise<KubernetesListObject<kind.Node>> {
+  return K8s(kind.Node).Get();
+}
+
+/**
+ * Update all NetworkPolicies labeled with uds/generated=KubeNodes to
+ * reflect the given node CIDRs.
+ */
+export async function updateKubeNodesNetworkPolicies(newNodes: V1NetworkPolicyPeer[]) {
+  const netPols = await K8s(kind.NetworkPolicy)
+    .WithLabel("uds/generated", RemoteGenerated.KubeNodes)
+    .Get();
+
+  for (const netPol of netPols.items) {
+    if (!netPol.spec) {
+      log.warn(
+        `KubeNodes NetworkPolicy ${netPol.metadata?.namespace}/${netPol.metadata?.name} is missing spec.`,
+      );
+      continue;
+    }
+
+    if (netPol.spec.egress) {
+      netPol.spec.egress[0] = netPol.spec.egress[0] || { to: [] };
+      netPol.spec.egress[0].to = newNodes;
+    }
```

Comment:
Let's make sure we account for ingress here as well, this caught us recently for the KubeAPI policies so you should be able to copy the logic from [there](https://github.com/defenseunicorns/uds-core/blob/970-all-nodes/src/pepr/operator/controllers/network/generators/kubeAPI.ts#L173-L194).

Reply:
added in 9b4bc342